tmd/Topology/persistent_properties.py

```diff
@@ -0,0 +1,181 @@
+"""Persistent properties classes
+"""
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+class PersistentProperty(ABC):
+    """Abstract class for persistent properties that are defined
+    on persistent components.
+    """
+    @abstractmethod
+    def get(self, component_start):
+        """Get component property"""
+
+    @abstractmethod
+    def infinite_component(self, component_start):
+        """Get property for infinite component"""
+
+
+class NoProperty(PersistentProperty):
+    """Function class for extracting a barcode without properties
+    """
+    def __init__(self, *_):
+        pass
+
+    def get(self, _):
+        """Returns empty list, does not contribute to component"""
+        return []
+
+    def infinite_component(self, _):
+        """Returns empty list, does not contribute to component"""
+        return []
+
+
+class PersistentAngles(PersistentProperty):
+    """Bifurcation angles per component
+    Args:
+        tree (Tree): A tree object
+        section_begs (np.ndarray): (N,) Int array of section node starts
+        section_ends (np.ndarray): (N,) Int array of section node ends
+        section_parents (np.ndarray): (N,) Int array of section parent ids
+        section_children (np.ndarray): (N,) Int array of section children ids
+    """
+    def __init__(self, tree, section_begs, _, section_parents, section_children):
```

Comment:
I'm not in favor of the empty input arguments, but I currently fail to identify a better alternative that preserves the generalization power of the current implementation.

Reply:
What is the purpose of this empty argument? Is it to have a specific method signature?

Reply:
yeah, I will remove all arguments apart from tree and I will cache the parent/children and section starts/ends calculation so that they can be accessed from tree (where they come from anyway) without recalculating.